### ISSUE TITLE
[Clang] `-Wreserved-identifier` warning should not be raised for user-defined literals

### ISSUE THREAD

**Issue by SunBlack** [llvm/llvm-project#99307]
We have a file with user-defined literals:

```cpp
#pragma once

#include <cstddef>

constexpr unsigned long long operator"" _KiB(unsigned long long value)
{
	return value * 1024LL;
}
//...
```

Clang raised here the warning: `identifier '_KiB' is reserved because it starts with '_' followed by a capital letter`. As the documentation of [user-defined literals](https://en.cppreference.com/w/cpp/language/user_literal) itself suggest the underscore, I think we have an exception here, where an underscore is fine.

**Comment by frederick-vs-ja** [comment]
The posted form is deprecated per [CWG2521](https://cplusplus.github.io/CWG/issues/2521.html). You can find that the cppreference pages are only suggesting the form without whitespace now.

On the other hand, `operator"" _KiB` is ill-formed, no diagnostic required because the token `_KiB` itself is used.

You should just switch to write `operator""_KiB`.

**Comment by SunBlack** [comment]
Ah good to know. In this case this could be simply adjusted by Clang Format as special case?
